source/extensions/filters/http/cache/cache_filter_config.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+#include <functional>
+
+#include "envoy/extensions/filters/http/cache/v3/cache.pb.h"
+
+#include "source/common/common/logger.h"
+#include "source/extensions/filters/http/cache/http_cache.h"
+#include "source/extensions/filters/http/common/pass_through_filter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Cache {
+
+using CacheFilterConfigPb = envoy::extensions::filters::http::cache::v3::CacheConfig;
+using CacheFilterConfigPbRef = std::reference_wrapper<const CacheFilterConfigPb>;
```

Comment:
Why do you need a reference_wrapper rather than just a plain reference here? It's only set at init-time.

Reply:
If leave just reference then in `cache_filter` following line `config_ = std::move(*cfg);` raises error (like copy constructor is deleted ...).
For this reason I have added `reference_wrapper`

Reply:
I think that you will not have nullptr but sort of empty config you will). Removed std::move.

Reply:
The HttpCachePtr of the original would become a nullptr after being std::moved from. Looks like a std::reference_wrapper's move is the same as a copy so it wouldn't hurt that one.